modules/client/src/main/java/org/apache/ignite/internal/client/tx/ClientTransaction.java

```diff
@@ -87,7 +84,9 @@ public void commit() throws TransactionException {
     /** {@inheritDoc} */
     @Override
     public CompletableFuture<Void> commitAsync() {
-        setState(STATE_COMMITTED);
+        if (!state.compareAndSet(STATE_OPEN, STATE_CLOSE)) {
```

Comment:
We can probably replace `AtomicInteger` with `AtomicBoolean` now since there are only 2 states.

Reply:
fixed. Thanks.